ape_safe/accounts.py

```diff
@@ -208,67 +201,141 @@ def prepare_transaction(self, txn: TransactionAPI) -> TransactionAPI:
         # NOTE: Need to override `AccountAPI` behavior for balance checks
         return self.provider.prepare_transaction(txn)
 
-    def call(self, txn: TransactionAPI, **call_kwargs) -> ReceiptAPI:  # type: ignore[override]
-        # NOTE: Override just to produce a more specific exception with better error message
-        try:
-            return super().call(txn, **call_kwargs)
-        except ContractLogicError as e:
-            if e.message.startswith("GS"):
-                raise SafeLogicError(e.message) from e
+    def _encode_signatures(self, signatures: Dict[AddressType, MessageSignature]) -> HexBytes:
+        # NOTE: Must order signatures in ascending order of signer address (converted to int)
+        def addr_to_int(a: AddressType) -> int:
+            return to_int(hexstr=a)
 
-            else:
-                raise e
+        return HexBytes(
+            b"".join(
+                signatures[signer].encode_rsv() for signer in sorted(signatures, key=addr_to_int)
+            )
+        )
+
+    def _safe_tx_exec_args(self, safe_tx: SafeTx) -> List:
+        return list(safe_tx._body_["message"].values())
+
+    def _preapproved_signature(
+        self, signer: Union[AddressType, BaseAddress, str]
+    ) -> MessageSignature:
+        # Get the Safe-style "preapproval" signature type, which is a sentinel value used to denote
+        # when a signer approved via some other method, such as `approveHash` or being `msg.sender`
+        # TODO: Link documentation for this
+        return MessageSignature(
+            v=1,  # Approved hash (e.g. submitter is approved)
+            r=b"\x00" * 12 + to_bytes(hexstr=self.conversion_manager.convert(signer, AddressType)),
+            s=b"\x00" * 32,
+        )
+
+    @handle_safe_logic_error()
+    def _impersonated_call(self, txn: TransactionAPI, **safe_tx_and_call_kwargs) -> ReceiptAPI:
+        safe_tx = self.create_safe_tx(txn, **safe_tx_and_call_kwargs)
+        safe_tx_exec_args = self._safe_tx_exec_args(safe_tx)
+        signatures = {}
+
+        # Bypass signature collection logic and attempt to submit by impersonation
+        # NOTE: Only works for fork and local networks
+        # TODO: Once it's a bit easier to set storage slots natively, use that to impersonate
```

Comment:
signing threshold is stored in slot 4, so we wouldn't need approveHash transactions if we set it to 1 and send the tx from one of the owners.